### PR TITLE
Releases dev guide

### PR DESCRIPTION
**Pull request by RossKen**
Given my ability to mess up releases - adding a dev guide to outline the steps for myself (and others)

I will update the image paths to master once this has been reviewed.

### CODE REVIEW

File: docs/dev_guides/changing_splink/releases.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Releasing a new version of Splink
+
+Splink is regularly updated with releases to add new features or bug fixes to the package.
```
Comment (ThomasHepworth) [inline docs/dev_guides/changing_splink/releases.md:3]: Something on semantic versioning?

Comment (ThomasHepworth) [reply docs/dev_guides/changing_splink/releases.md:3]: A link is sufficient here.

**Review by ThomasHepworth — CHANGES_REQUESTED** [review]
Minor comments and then I'll approve.

File: docs/dev_guides/changing_splink/releases.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Below are the steps for releasing a new version of Splink:
+
+1. Update [**pyproject.toml**](https://github.com/moj-analytical-services/splink/blob/master/pyproject.toml) and [**__init__.py**](https://github.com/moj-analytical-services/splink/blob/master/splink/__init__.py) with the latest version.
+2. Generate a new release in the [releases section of the repo](https://github.com/moj-analytical-services/splink/releases)
+
+Including:
+
+- Choosing a new release tag (which matches your updates to [**pyproject.toml**](https://github.com/moj-analytical-services/splink/blob/master/pyproject.toml) and [**__init__.py**](https://github.com/moj-analytical-services/splink/blob/master/splink/__init__.py). Ensure that your release tag follows [semantic versioning](https://docs.npmjs.com/about-semantic-versioning).
+
+![](https://raw.githubusercontent.com/moj-analytical-services/splink/master/docs/img/releases-tag.png))
```
[the remark this answers is not in the document]
Comment (RossKen) [reply docs/dev_guides/changing_splink/releases.md:14]: The images? I have switched them to point at master (i.e. not this branch) for when it is merged in. Apologies, I should have left a comment to make that clear

**Review by ThomasHepworth — APPROVED** [review]

**Pull request merged**